File: pubnub-java/src/main/java/com/pubnub/internal/endpoints/TimeImpl.java
package com.pubnub.internal.endpoints;

import com.pubnub.api.endpoints.Time;
import com.pubnub.api.models.consumer.PNTimeResult;
import com.pubnub.internal.InternalPubNubClient;

public class TimeImpl extends DelegatingEndpoint<PNTimeResult>implements Time {

    public TimeImpl(InternalPubNubClient pubnub) {
        super(pubnub);
    }

    @Override
    protected com.pubnub.internal.Endpoint<?, PNTimeResult> createAction() {
        return pubnub.time();
    }
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jsMain/kotlin/com/pubnub/api/endpoints/TimeImpl.kt
package com.pubnub.kmp.endpoints

import PubNub
import com.pubnub.api.EndpointImpl
import com.pubnub.api.models.consumer.PNTimeResult

class TimeImpl(pubnub: PubNub) : Time, EndpointImpl<PubNub.FetchTimeResponse, PNTimeResult>(
    promiseFactory = { pubnub.time() },
    responseMapping = { PNTimeResult(it.timetoken.toLong()) }
)